libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseLexer.g4

```diff
@@ -368,7 +368,7 @@ STRING
     ;
 
 ESCAPED_STRING
-    : 'E' '\'' ( ~'\'' | '\'\'' | '\\\'' )* '\''
+    : 'E' '\'' ( ~('\'' | '\\') | '\'\'' | ('\\'+  ~'\\') )* '\''
```

Comment:
Could you please explain a bit this?
1. `~('\'' | '\\')` we don't allow single quote or single backslash, correct?
2. `'\'\''` we allow an open-closing single quote (without anything in between), correct?
3. ('\\'+ ~'\\') I don't get .... :(

Reply:
1. Yes. Extended answer - we actually allow both `\` and `'` inside  `E''` but only specific combinations, that's why we disable standalone matches, in order to be able to specify more precisely what we want. 

2. Specific combination for matching `'`. We allow escaping quote by another quote, so `E'abc'''` is a a valid string producing `abc'`. 

3. Specific combination for matching `\`. `'\\'+` stands for one or more slashes followed by a non-slash symbol `~'\\'`

Reply:
great, thx a lot!
Would you mind adding some comments there?

Reply:
Added in https://github.com/crate/crate/pull/15230/commits/6c9d21e4188cb599243cbe86ce56d49cd817685b